src/backend/distributed/transaction/backend_data.c

```diff
@@ -400,6 +406,8 @@ InitializeBackendData(void)
 
 	Assert(MyBackendData);
 
+	LockBackendSharedMemory(LW_SHARED);
```

Comment:
But if we do it exclusive, we'd prevent concurrent backend initializations.  That's why we keep here shared and acquire the exclusive one forming the graph?